### DISCUSSION TITLE
Show/Hide fields on form based on values from REST query from a table.

### DISCUSSION THREAD

**Discussion by tokkisva** [Budibase/budibase#8528] [Help] [answered]
Hi all!

I was wondering if there was a way to show or hide fields based on a value not from the form fields- but rather from a Data Provider query displayed in a table. 

I have a field that, when a value is entered- it executes a GET to a REST API and populates some tables on the form. 

In field 'ASN ID #', whatever value is entered there drives the GET request to the REST API data source linked to the table with a binding. That REST GET displays data for that record in the tables shown in screenshot. 

I want to show/hide fields based on the output from that query that populates in the table. I'd also like to pre-fill some values on the form that are obtained from the REST GET that occurs after the form field ASN ID # is filled out. 

Example: if the PO NUMBER starts with 'somevalue', hide form field 'x'.
Example2: pre-fill the LOBID form field with the LOBID result from the REST GET.

![image](https://user-images.githubusercontent.com/38924262/200075646-4e95f5fd-1610-4811-afec-1800174450ee.png)

Is this possible? Can you call a table from data source in javascript perhaps?

Thanks in advance!

**Comment by melohagan** [comment, marked as answer]
Hey Kyle,

Have you had a look at the conditional UI? https://docs.budibase.com/docs/conditions
You can apply conditional show/hide to form fields.

So consider this component tree setup:
![Screenshot 2022-11-07 at 13 07 45](https://user-images.githubusercontent.com/101575380/200317998-59883ffc-c02e-4c32-9682-c06170ab2fae.png)

You can then add this condition to the *Notes* form field:
![Screenshot 2022-11-07 at 13 08 15](https://user-images.githubusercontent.com/101575380/200318084-9caa6365-8042-4c3b-8ad7-03c9de9ddd35.png)

**Comment by tokkisva** [comment]
Hi Melohagan,

Seems like I might need to change from a table to repeater to create a binding. 
I'm going through that process now but when I add in the PO Number to the repeater paragraph text binding, I get this error:
Current Handlebars syntax is invalid, please check the guide [here](https://handlebarsjs.com/guide/) for more details.

Is this because the API GET is already using handlebars to drive the query?

![image](https://user-images.githubusercontent.com/38924262/200354193-8e1963d6-7742-4d10-9a27-5961729e3dff.png)

Here's how the data source is setup, it executes a query based on what is entered in field OP1:
![image](https://user-images.githubusercontent.com/38924262/200354576-97047821-1b65-455c-b2e6-933c8c6a5970.png)

Field OP1 (drives the GET for the data source in the repeater):
![image](https://user-images.githubusercontent.com/38924262/200354654-e6b4f5cc-4c92-46ab-9eb4-730a8f257875.png)

**Comment by tokkisva** [comment]
Hey I actually got this to work!! I was able to make a binding for conditions using the table with this sort of syntax:
{{ ASN Data.Rows.0.poNo}}

Thank you so much for pointing me in the right direction!

**Reply by melohagan** [reply]
Awesome! Glad it worked :)
